File: 程式設計：資料結構答案檔/C#/402.cs
using System;

class Program
{
    static void Main()
    {
        string[] input = Console.ReadLine().Split(' ');
        int m = int.Parse(input[0]), n = int.Parse(input[1]);

        int[,] adjMatrix = new int[m+1, m+1];
        for (int i = 0; i < n; i++)
        {
            input = Console.ReadLine().Split(' ');
            int a = int.Parse(input[0]), b = int.Parse(input[1]);
            adjMatrix[a, b] = 1;
            adjMatrix[b, a] = 1;
        }

        int[] degree = new int[m+1];
        for (int i = 1; i <= m; i++)
            for (int j = 1; j <= m; j++)
                if (adjMatrix[i, j] == 1)
                    degree[i]++;

        int numOfOddDegree = 0;
        for (int i = 1; i <= m; i++)
            if (degree[i] % 2 == 1)
                numOfOddDegree++;

        Console.WriteLine(numOfOddDegree == 0 || numOfOddDegree == 2 ? "YES" : "NO");

        for (int i = 1; i <= m; i++)
        {
            for (int j = 1; j < m; j++)
                Console.Write(adjMatrix[i, j] + " ");
            Console.WriteLine(adjMatrix[i, m]);
        }
    }
}

File: 程式設計：資料結構答案檔/Go/402.go
package main

import (
	"fmt"
)

func main() {
	var m, n int
	fmt.Scan(&m, &n)

	adjMatrix := make([][]int, m+1)
	for i := range adjMatrix {
		adjMatrix[i] = make([]int, m+1)
	}

	for i := 0; i < n; i++ {
		var a, b int
		fmt.Scan(&a, &b)
		adjMatrix[a][b] = 1
		adjMatrix[b][a] = 1
	}

	degree := make([]int, m+1)
	for i := 1; i <= m; i++ {
		for j := 1; j <= m; j++ {
			if adjMatrix[i][j] == 1 {
				degree[i]++
			}
		}
	}

	numOfOddDegree := 0
	for i := 1; i <= m; i++ {
		if degree[i]%2 == 1 {
			numOfOddDegree++
		}
	}

	if numOfOddDegree == 0 || numOfOddDegree == 2 {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}

	for i := 1; i <= m; i++ {
		for j := 1; j < m; j++ {
			fmt.Printf("%d ", adjMatrix[i][j])
		}
		fmt.Println(adjMatrix[i][m])
	}
}
